authorization/invitation/service/invitation_service.go

```diff
@@ -331,3 +332,85 @@ func lookupSpaceName(ctx context.Context, witURL string, spaceID string) (string
 
 	return *spaceSingle.Data.Attributes.Name, nil
 }
+
+// Accept processes an invitation acceptance click, and returns the resource ID of the resource or identity resource which the invitation is for
+func (s *invitationServiceImpl) Accept(ctx context.Context, currentIdentityID uuid.UUID, token uuid.UUID) (string, error) {
+	var resourceID string
+
+	// Locate the invitation
+	inv, err := s.Repositories().InvitationRepository().FindByAcceptCode(ctx, currentIdentityID, token)
+
+	if err != nil {
+		return resourceID, err
+	}
+
+	// If this invitation is for an identity
+	if inv.InviteTo != nil {
+		inviteToIdentity, err := s.Repositories().Identities().Load(ctx, *inv.InviteTo)
+		if err != nil {
+			return resourceID, err
+		}
+
+		// If the invitation is for a membership, add a membership record
+		if inv.Member {
+			s.Repositories().Identities().AddMember(ctx, inviteToIdentity.ID, currentIdentityID)
+		}
+
+		roles, err := s.Repositories().InvitationRepository().ListRoles(ctx, inv.InvitationID)
+		if err != nil {
+			return resourceID, err
+		}
+
+		// If the invitation includes role assignments, assign them
+		for _, role := range roles {
+			ir := &repository.IdentityRole{
+				IdentityID: currentIdentityID,
+				RoleID:     role.RoleID,
+				ResourceID: inviteToIdentity.IdentityResourceID.String,
+			}
+
+			err = s.Repositories().IdentityRoleRepository().Create(ctx, ir)
+			if err != nil {
+				return resourceID, err
+			}
+		}
+
+		// Delete the invitation
+		s.Repositories().InvitationRepository().Delete(ctx, inv.InvitationID)
```

Comment:
Since a user could have been sent out multiple invites, this needs to delete "All invites to user X who have been invited with Roles Y,Z...". This is to ensure that every other invite mail should lead to "this invite is no longer valid because..."

Reply:
The accept invitation request passes a single acceptance code, which itself belongs to a single invitation.  Once the invitation has been accepted, that record alone needs to be removed, there are no circumstances under which we should be deleting any other invitations here.

Reply:
On a related note however, we do need to be deleting the `invitation_role` records - I just checked and it's not being done.  I'll fix that up in the invitation repository.

Reply:
> The accept invitation request passes a single acceptance code, which itself belongs to a single invitation. Once the invitation has been accepted, that record alone needs to be removed, there are no circumstances under which we should be deleting any other invitations here.

If multiple (re)invites were sent to the same user for the same set of roles, and one of them was accepted by the user - how do we ensure we don't land up with many many unaccepted invites ?

Reply:
If a reinvite is issued for the same team, org or resource, then the same invitation should be sent again.  In fact, we should enforce this via a database constraint, so we should probably create a new issue for this.

Reply:
Another thing we don't have is a "Cancel Invitation" endpoint, we likely need that also.  I'll create a new issue for this, and the DB constraint.

Reply:
Created #543

Reply:
Having it as a db constraint is a good idea. One advantage of allowing multiple invitations in the db is that it helps us record the number of (re)invitations that were created. If you feel we should go with the db constraint, I'm with you on that. Later on, if the need for recording sent invites shows up, we can fix that.

Reply:
Yes, I believe that a reinvite should simply send another notification for the existing invitation.

Reply:
Cool, let's roll with that, then :+1: